docs/schemaValidationMiddleware.md

```diff
@@ -0,0 +1,126 @@
+# The Schema "Gatekeeper" Validation Middleware
+
+Although you usually will build some fault tolerance into your components, on some level they expect to deal with
```

Comment:
It's a fair point, I have revised the wording to more clearly reflect the limits of this type of validation.